sn_networking/src/event/swarm.rs

```diff
@@ -137,7 +137,8 @@ impl SwarmDriver {
                                 our_protocol: IDENTIFY_PROTOCOL_STR.to_string(),
                                 their_protocol: info.protocol_version,
                             });
-                            self.remove_outdated_connections(Some(peer_id));
+                            // Block the peer from any further communication.
+                            self.swarm.behaviour_mut().blocklist.block_peer(peer_id);
```

Comment:
so the already established connection will be disconnected automatically, once the peer is in the blocklist?
i.e. no more explicit `remove_connection` call to that peer is needed ?

Reply:
That is right! It internally calls `ToSwarm::CloseConnection`